src/control_plane/worker_stats/worker.ts

```diff
@@ -130,8 +137,61 @@ class Worker {
 
     this.#registerTime = Date.now();
 
+    this.#initializationTimeout = initializationTimeout ?? config.worker.defaultInitializerTimeout;
+
     this.logger = new PrefixedLogger('worker_stats worker', this.#name);
     this.requestId = undefined;
+
+    this.#readyDeferred = createDeferred<void>();
+    this.#pendingReady = undefined;
+  }
+
+  async ready() {
+    this.#pendingReady = true;
+
+    // 在 await ready 之前已经 Ready 了
+    if (this.#containerStatus >= ContainerStatus.Ready) {
+      this.logger.info('Worker(%s, %s) already ready before pending ready', this.#name, this.#credential);
+      this.#pendingReady = false;
+      return this.#readyDeferred.resolve();
```

Comment:
这里应该返回 `deferred.promise` 吧

```suggestion
      this.#readyDeferred.resolve();
      return this.#readyDeferred.promise;
```